lib/ecto/query/builder.ex

```diff
@@ -121,6 +121,11 @@ defmodule Ecto.Query.Builder do
     escape_with_type(op_expr, type, params_acc, vars, env)
   end
 
+  def escape({:type, _, [{bitwise_op, _, [_, _]} = op_expr, type]}, _type, params_acc, vars, env)
+      when bitwise_op in ~w(&&& |||)a do
```

Comment:
```suggestion
      when bitwise_op in ~w(&&& ||| <<< >>>)a do
```
It looks like Postgres also supports the shift operators: https://www.postgresql.org/docs/current/functions-bitstring.html

But note we also need to make changes to Ecto SQL so the adapter can convert these operators into the correct query string (i.e. convert `&&&` to `&`): https://github.com/elixir-ecto/ecto_sql/blob/master/lib/ecto/adapters/postgres/connection.ex#L420

Similarly for the unary operators we'd need to add function head like this: https://github.com/elixir-ecto/ecto_sql/blob/master/lib/ecto/adapters/postgres/connection.ex#L875

So that `~~~` works. We can probably do the same for the bitwise functions `band`, `bor`, `bnot`, `bsl`, `bsr`, and `bxor`. Since there is no binary operator for xor this would give users a clean way to use Postgres's xor operator.

Reply:
Hi @greg-rychlewski, thank you very much for your review.

I'm working on a companion ecto_sql PR that will convert everything into a query string too.